acr.tf

```diff
@@ -0,0 +1,32 @@
+resource "azurerm_resource_group" "acr" {
+  name     = local.resource_prefix
+  location = local.azure_location
+  tags     = local.tags
+}
+
+resource "azurerm_container_registry" "acr" {
+  name                          = replace(local.resource_prefix, "-", "")
+  resource_group_name           = azurerm_resource_group.acr.name
+  location                      = azurerm_resource_group.acr.location
+  sku                           = local.registry_sku
+  admin_enabled                 = local.registry_admin_enabled
+  public_network_access_enabled = local.registry_public_access_enabled
+  tags                          = local.tags
+
+  dynamic "retention_policy" {
+    for_each = local.registry_sku == "Premium" ? [1] : []
+
+    content {
+      days    = local.registry_retention_days
+      enabled = local.enable_registry_retention_policy
+    }
+  }
+
+  dynamic "network_rule_set" {
+    for_each = local.registry_sku == "Premium" ? [1] : []
+
+    content {
+      default_action = "Deny"
```

Comment:
This will block any `push` events from GitHub unless the client is using a Private Endpoint. Is this what we want? Maybe worth setting this to allow for the time being seeing as though we can't really restrict it

Reply:
I can add to the condition `&& !local.registry_public_access_enabled`, which would then have the expected result

We could then add to it to allow adding IP rules